### ISSUE TITLE
doesn't work with any version of atom I've tried

### ISSUE THREAD

**Issue by ryran** [Stepsize/atom-better-git-blame#33]
I love the screenshots, and am excited at the prospect of the enhancements over the simpler git-blame package I use currently, but after installing from atom's UI (not from source), this package doesn't ever work for me; it just opens an empty pane on the left side.

![better-git-blame1](https://user-images.githubusercontent.com/1092618/39599938-442ae816-4eeb-11e8-8632-481f0662cd22.png)
![better-git-blame2](https://user-images.githubusercontent.com/1092618/39599941-47d42108-4eeb-11e8-9e93-52d2b1f0fa3e.png)

I'm not sure how to troubleshoot it, so any ideas would be greatly appreciated.

**Comment by nomeyer** [comment]
Hey @ryran, thanks for getting in touch! Glad you're excited 😁

I just tried the latest version of the plugin (`0.4.5`) on the latest version of Atom (`1.26.1`) and it all worked fine for me.

To help me troubleshoot, could you:
- Let me know which version of Atom you are running on which OS
- Try again with the Atom console open (⌥+⌘+i or View > Developer > Toggle Developer Tools) and report any logs you see when trying out the plugin
- Potentially try using it on other repos

Thanks!

**Comment by ryran** [comment]
> Let me know which version of Atom you are running on which OS

Well I've been trying it every couple weeks for the last few months and it's never worked. I always run the latest beta as soon as it comes out, so I'm on `1.27.0-beta1` right now. Linux (Fedora) x64. What prompted me to finally open an issue: I just reinstalled a fresh machine and it failed right out of the box so I finally started to think it had nothing to do with my existing setup.

> Try again with the Atom console open (⌥+⌘+i or View > Developer > Toggle Developer Tools) and report any logs you see when trying out the plugin

![bgb-console-errs](https://user-images.githubusercontent.com/1092618/39720963-f6a5549a-520b-11e8-911f-630b0f7defed.png)

> Potentially try using it on other repos

Same result in all repos.

**Comment by ryran** [comment]
I just now bothered to google about this and figured out what it is. https://github.com/atom/atom/issues/13451

In short, the official Atom RPMs  (Fedora/RHEL linux) provide a `/usr/share/applications/atom.desktop` or `/usr/share/applications/atom-beta.desktop` file. These are simple text files that describe an app-launcher (shortcut) icon.

Atom is (IMHO) incorrectly calling calling the `atom`/`atom-beta` commands via `/usr/share/atom/atom` or `/usr/share/atom-beta/atom` when they should be using `/usr/bin/atom` or `/usr/bin/atom-beta`.

If you execute atom from the latter, you don't run into any problems. If you execute atom via the officially-provided shortcut/launcher, you get errors and problems as described above or in the linked issue (and issues linked from it). I also realize now that this is why I couldn't get some python linting packages to work.

Long story short: it's not your fault and not your problem .. and I can perform a work-around, but the fix should come from atom itself. Unfortunately, if lee-dohm's ignorant & insular response in the above-linked issue is any indication... that's not gonna happen any time soon.

**Comment by ryran** [comment]
And may I just say: now that I can finally see it .... your package is amazing!

**Comment by nomeyer** [comment]
Thanks a lot for looking into this @ryran and sharing your workaround!

**Comment by ryran** [comment]
For the record, the atom core team has attempted to address the original issue in a way via https://github.com/atom/atom/pull/17508 such that my original work-around is no longer possible (the application launcher calls `/usr/bin/atom` or `/usr/bin/atom-beta` now); however, between this change and some subsequent stuff (probably https://github.com/atom/atom/pull/19138), the issue still remains.

Sad.

![git-blame-sad](https://user-images.githubusercontent.com/1092618/71736768-eccdde00-2e1f-11ea-9bdd-2066bda6ca92.png)

The work-around is to launch atom from a terminal. Sigh.

**Comment by nomeyer** [comment]
Hey @ryran, thanks for reporting this.

I'm really sorry but unfortunately we no longer support this Atom plugin... If you submit a pull request that fixes your issue we'll be sure to merge it in though!

All the best
